src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int STOCK_ARR_LENGTH = 16;
+    private static final float MAX_FILL = 0.75f;
+    private static final int GROWTH_COEFICIENT = 2;
+    private Node[] arr;
+    private int size;
+
+    public MyHashMap() {
+        this.arr = new Node[STOCK_ARR_LENGTH];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int pos = bucketFor(key);
+        if (arr[pos] == null) {
+            arr[pos] = new Node<>(key, value);
+            increaseSize();
+            return;
+        }
+        Node tempNode = findNode(key);
+        if (keysAreEqual(key, (K)tempNode.key)) {
+            tempNode.value = value;
+            return;
+        }
+        tempNode.next = new Node<>(key, value);
+        increaseSize();
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node tempNode = findNode(key);
+        return arr[bucketFor(key)] == null || !keysAreEqual(key, (K)findNode(key).key)
+                ? null : (V)tempNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int bucketFor(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % arr.length;
+    }
+
+    private Node findNode(K key) {
```

Comment:
It is confusing because I expect to get node by key, but I can get the last node also.  Think about how you can improve it. Maybe naming can resolve this problem